File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_CSharp/SceneObject.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: SceneObject.cs
//
//--------------------------------------------------------------------------

namespace Microsoft.ParallelComputingPlatform.ParallelExtensions.Samples
{
    abstract class SceneObject
    {
        public Surface Surface;
        public abstract ISect Intersect(Ray ray);
        public abstract Vector Normal(Vector pos);

        public SceneObject(Surface surface) { Surface = surface; }
    }
}

File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_VisualBasic/SceneObject.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: SceneObject.vb
'
'--------------------------------------------------------------------------

Public MustInherit Class SceneObject
    Public Surface As Surface
    Public MustOverride Function Intersect(ByVal ray As Ray) As ISect
    Public MustOverride Function Normal(ByVal position As Vector) As Vector

    Public Sub New(ByVal surface As Surface)
        Me.Surface = surface
    End Sub
End Class